provider/runbook_data.go

```diff
@@ -53,10 +65,16 @@ func dataFireHydrantRunbook(ctx context.Context, d *schema.ResourceData, m inter
 		return diag.Errorf("Error reading runbook %s: %v", runbookID, err)
 	}
 
+	attachmentRule, err := json.Marshal(runbookResponse.AttachmentRule)
+	if err != nil {
+		return diag.Errorf("Error converting step config to JSON due invalid JSON returned by FireHydrant: %v", err)
```

Comment:
```suggestion
		return diag.Errorf("Error converting attachment rule to JSON due invalid JSON returned by FireHydrant: %v", err)
```

Reply:
also, are we guaranteed to always have an attachment rule?

Reply:
oh you are probably right will move it below and check if AttachmentRule is not nil first
```golang

if runbookResponse.AttachmentRule != nil {
	attachmentRule, err := json.Marshal(runbookResponse.AttachmentRule)
	if err != nil {
		return diag.Errorf("Error converting attachment_rule to JSON due invalid JSON returned by FireHydrant: %v", err)
	}
	attributes["attachment_rule"] = string(attachmentRule)
}

```